### ISSUE TITLE
GameNetworkingSockets bindings

### ISSUE THREAD

**Issue by VinTarZ** [LWJGL/lwjgl3#776]
### Library

GameNetworkingSockets

### Project page

https://github.com/ValveSoftware/GameNetworkingSockets

### Source repository

https://github.com/ValveSoftware/GameNetworkingSockets

### Platforms

Linux x64, Windows x64, Windows x86

### Description

GameNetworkingSockets seems like all-in-one solution for making multiplayer games when TCP networking is not good enough:

> Reliable & unreliable messages over UDP. Robust message fragmentation & reassembly. P2P networking / NAT traversal. Encryption.

**Comment by Sm0keySa1m0n** [comment]
Coincidentally I just came to this issue tracker to request the exact same thing. This library seems to be a great solution for game networking and is very well maintained so it would be great to see this in LWJGL. There's hardly any equivalent libraries out there for Java at the moment, especially ones that are actively maintained.

**Comment by ws909** [comment]
[BUILDING.md](https://github.com/ValveSoftware/GameNetworkingSockets/blob/master/BUILDING.md) states that it can build on MacOS, but you didn't include it in the list of supported platforms? How come?

**Comment by philipguin** [comment]
Yes please! I’ve been thinking about doing JNI bindings manually for my game, this would be a godsend!
